My highlights button and pop up

KarinaMendez2 opened the pull request:
Initial styles for user not logged in for my highlights pop up, created selectors for my highlights opened and closed actions

Commit:
Initial styles for user not logged
in for my highlights pop up,
created selectors for my
highlights opened and closed
actions

TomWoodward committed:
rearrange styling of sticky notes

Commit:
moved actions and selectors
from content to highlights,
added close button, sticky
notes fixes, addes login link
to html translation, added logged in user highlights state in popup

Commit:
updated snaps and initial tests
for my highlights popup

Commit:
merge conflicts

Commit:
lint errors fixed

TomWoodward committed:
Merge branch 'master' into my_highlights_button

TomWoodward commented on src/app/components/Layout.tsx:
```diff
@@ -10,6 +11,7 @@ import SkipToContentWrapper from './SkipToContentWrapper';
 const Layout: SFC = ({ children }) => <SkipToContentWrapper>
   <NavBar />
   <ErrorModal />
+  <HighlightsPopUp />
```
this should be in `/src/app/content/components/Content.tsx`

TomWoodward commented on src/app/content/components/Toolbar/HighlightsPopUp.tsx:
```diff
@@ -0,0 +1,441 @@
+                </FormattedMessage>
+                <CloseIcon data-testid='close-highlights-popup' onClick={() => this.props.closeMyHighlights()}/>
+              </Header>
+              {this.props.user ? this.myHighlights() : this.loginForHighlights()}
```
we should split the `myHighlights` and `loginForHighlights` content into separate files, this file is getting really big already.

TomWoodward commented on src/app/content/components/Toolbar/index.spec.tsx:
```diff
@@ -0,0 +1,79 @@
+import HighlightButton from './HighlightButton';
+import HighlightsPopUp from './HighlightsPopUp';
+
+describe('MyHighlights', () => {
```
this file has tests for both the highlights button and the modal popup, neither of those tests really have anything to do with the toolbar but the test file is called `Toolbar/index.spec.tsx`.

i think these tests look good, but they should be separated out for the components that they're testing.

TomWoodward commented on src/app/content/reducer.ts:
```diff
@@ -17,6 +17,7 @@ import { getPageSlug } from './utils/archiveTreeUtils';
 export const initialState = {
   highlights: initialHighlightState,
   loading: {},
+  myHighlightsOpen: initialHighlightState.myHighlightsOpen,
```
this is already dealt with in `src/app/content/highlights/reducer.ts`, you don't need it here

TomWoodward commented on src/app/content/types.ts:
```diff
@@ -11,6 +11,7 @@ export interface Params {
 
 export interface State {
   tocOpen: boolean | null;
+  myHighlightsOpen: boolean;
```
or here

Commit:
Merge remote-tracking branch 'origin/master' into my_highlights_button

Commit:
moving components around to where
they're supposed to be, separated
my highlights button and popup styles
and specs into folder, removed unused
declarations

TomWoodward commented on src/app/content/components/Content.tsx:
```diff
@@ -167,6 +168,7 @@ const Content = ({mobileExpanded}: {mobileExpanded: boolean}) => <Layout>
   />
   <Background>
     <BookBanner />
+    <HighlightsPopUp />
```
might as well put this inside the error boundary one line down

TomWoodward commented on src/app/content/components/Toolbar/MyHighlights/HighlightStyles.tsx:
```diff
@@ -0,0 +1,297 @@
+import React from 'react';
```
please move all the modal stuff out of `Toolbar` and into `content/highlights/components`

TomWoodward commented on src/app/content/components/Toolbar/MyHighlights/HighlightStyles.tsx:
```diff
@@ -0,0 +1,297 @@
+    height: 8.8,
+    width: 18,
+  },
+  width: 24.8 + 5, /* to allow text to fit in one line with tooltip */
```
why not just set this to `29.8`? why do the math here like this?

TomWoodward commented on src/app/content/components/Toolbar/MyHighlights/HighlightsPopUp.tsx:
```diff
@@ -0,0 +1,146 @@
+class HighlightsPopUp extends Component<Props> {
+  public loginForHighlights = () => {
+    return <Styled.PopupBody>
+      <Styled.LoginText values={{loginLink: '/accounts/login?r=' + this.props.currentPath}} />
```
i added `loginLink` to authSelectors so you don't have to have `'/accounts/login?r='` hardcoded here anymore

TomWoodward commented on src/app/content/highlights/reducer.ts:
```diff
@@ -24,6 +25,10 @@ const reducer: Reducer<State, AnyAction> = (state = initialState, action) => {
+    case getType(actions.openMyHighlights):
+      return {...state, myHighlightsOpen: true};
+    case getType(actions.closeMyHighlights):
+      return {...state, myHighlightsOpen: false};
```
should probably also set `myHighlightsOpen: false` on `locationChange` up on line 23

Commit:
Merge remote-tracking branch 'origin/master' into my_highlights_button

Commit:
moved highlight pop up stuff to
highlight folder, removed hardcoded
login link, updated imports and snap

Commit:
Merge remote-tracking branch 'origin/master' into my_highlights_button

Commit:
re ordered texts on sticky
notes and modified other
translations

Commit:
attempt to focus on highlights
popUp

Commit:
Merge remote-tracking branch 'origin/master' into my_highlights_button

Commit:
added test for focus when
pop up is open

Commit:
Merge remote-tracking branch 'origin/master' into my_highlights_button

TomWoodward committed:
fix test

TomWoodward committed:
Merge branch 'my_highlights_button' of github.com:openstax/books-web into my_highlights_button

Commit:
re formatted highlights popup file, removed array in return elements

TomWoodward commented on src/app/content/highlights/components/HighlightStyles.tsx:
```diff
@@ -0,0 +1,297 @@
+
+// tslint:disable-next-line:variable-name
+export const Header = styled(H3)`
+  background: #002569;
```
this is supposed to match the book color, don't have to change it now but you should make a follow up PR before we forget

TomWoodward committed:
fix lighthouse

TomWoodward committed:
Merge branch 'my_highlights_button' of github.com:openstax/books-web into my_highlights_button

TomWoodward committed:
:camera:

TomWoodward committed:
hide highlights button unless feature flag is on
